meetings/Meetings.js

```diff
@@ -0,0 +1,358 @@
+    rows.unshift(header);
+
+    const spreadsheet = SpreadsheetApp.openById(getReportSheetId_());
+    const sheet = SheetUtil.ensureSheet(spreadsheet, "1on1recurring");
+    sheet.getRange(1, 1, sheet.getMaxRows(), header.length).clearContent();
+    sheet.getRange(1, 1, rows.length, header.length).setValues(rows);
+}
+
+
+/** Utility function to visit and handle all personal employee Gcal events.
+ *
+ * @param visitor The visitor function which receives (event, employee, employees, calendar, personio) as arguments and may return false to stop iteration.
+ * @param listParams Additional parameter overrides for calendar.list().
+ */
+async function visitEvents_(visitor, listParams) {
+
+    const allowedDomains = (getScriptProperties_().getProperty(ALLOWED_DOMAINS_KEY) || '')
+        .split(',')
+        .map(d => d.trim());
+
+    const emailWhiteList = getEmailWhiteList_();
+    const isEmailAllowed = email => (!emailWhiteList.length || emailWhiteList.includes(email))
+        && allowedDomains.includes(email.substring(email.lastIndexOf('@') + 1));
+
+    Logger.log('Configured to handle accounts %s on domains %s', emailWhiteList.length ? emailWhiteList : '', allowedDomains);
+
+    // all timing related activities are relative to this EPOCH
+    const epoch = new Date();
+
+    // how far back to sync events/time-offs
+    const lookbackMillies = -Math.round(getLookbackDays_() * 24 * 60 * 60 * 1000);
+    // how far into the future to sync events/time-offs
+    const lookaheadMillies = Math.round(getLookaheadDays_() * 24 * 60 * 60 * 1000);
+
+    const fetchTimeMin = Util.addDateMillies(new Date(epoch), lookbackMillies);
+    fetchTimeMin.setUTCHours(0, 0, 0, 0); // round down to start of day
+    const fetchTimeMax = Util.addDateMillies(new Date(epoch), lookaheadMillies);
+    fetchTimeMax.setUTCHours(24, 0, 0, 0); // round up to end of day
+
+    const personioCreds = getPersonioCreds_();
+    const personio = PersonioClientV1.withApiCredentials(personioCreds.clientId, personioCreds.clientSecret);
+
+    // load and prepare list of employees to process
+    const employees = await personio.getPersonioJson('/company/employees').filter(employee =>
+        employee.attributes.status.value !== 'inactive' && isEmailAllowed(employee.attributes.email.value)
+    );
+
+    Logger.log('Visiting events between %s and %s for %s accounts', fetchTimeMin.toISOString(), fetchTimeMax.toISOString(), '' + employees.length);
+
+    let firstError = null;
+    let processedCount = 0;
+    let done = false;
+    for (const employee of employees) {
+
+        const email = employee.attributes.email.value;
+
+        // we keep operating if handling calendar of a single user fails
+        try {
+            const calendar = await CalendarClient.withImpersonatingService(getServiceAccountCredentials_(), email);
+            const allEvents = await calendar.list('primary', {
+                singleEvents: false,
```

Comment:
What does singleEvents mean in this case?

Reply:
Added comments.

It means that recurring events aren't rolled out into individual instances (we wan't the original objects in this case).